### PR TITLE
Fix `OPTForQuestionAnswering` doctest

### PR DESCRIPTION
**Pull request by ydshieh**
# What does this PR do?

The checkpoint has no QA head, so we need to set seed and change the expected values.

### CODE REVIEW

File: src/transformers/models/opt/modeling_opt.py [hunk 1]
```diff
@@ -1173,7 +1158,36 @@ def forward(
...
+        >>> with torch.no_grad():
+        ...     outputs = model(**inputs)
+
+        >>> answer_start_index = outputs.start_logits.argmax()
+        >>> answer_end_index = outputs.end_logits.argmax()
+
+        >>> predict_answer_tokens = inputs.input_ids[0, answer_start_index : answer_end_index + 1]
+        >>> predicted = tokenizer.decode(predict_answer_tokens)
+        >>> predicted
+        ' Henson?'
```
Comment (ydshieh) [inline src/transformers/models/opt/modeling_opt.py:1189]: Due to the leading space in the outputs, I am not able to simply doing
```python
tokenizer.decode(predict_answer_tokens)
 Henson?   --> (here with leading space)
```
as the test will think the expected values being `Henson?` (without space).

**Review by sgugger — APPROVED** [review]
Thanks for fixing!

File: src/transformers/models/opt/modeling_opt.py [hunk 2]
```diff
@@ -1173,7 +1158,36 @@ def forward(
...
+        >>> import torch
+
+        >>> torch.manual_seed(4)  # doctest: +IGNORE_RESULT
+        >>> tokenizer = GPT2Tokenizer.from_pretrained("facebook/opt-350m")
+
+        >>> # note: we are loading a OPTForQuestionAnswering from the hub here,
+        >>> # so the head will be randomly initialized, hence the predictions will be random
+        >>> model = OPTForQuestionAnswering.from_pretrained("facebook/opt-350m")
+
+        >>> question, text = "Who was Jim Henson?", "Jim Henson was a nice puppet"
```
Comment (ArthurZucker) [inline src/transformers/models/opt/modeling_opt.py:1177]:
```suggestion
        >>> question = "Who was Jim Henson?"
        >>> text =  "Jim Henson was a nice puppet"
```
Would just change that for clarity 😉

**Review by ArthurZucker — COMMENTED** [review]
LGTM, would just not do the 1 line initialisation

File: src/transformers/models/opt/modeling_opt.py (hunk 2 above)
Comment (ydshieh) [reply src/transformers/models/opt/modeling_opt.py:1177]: Well, it is taken from `PT_QUESTION_ANSWERING_SAMPLE` in `src/transformers/utils/doc.py`. I would like to keep as it is at this moment, and we can have a discussion to change to 2 lines for all occurrences later.

Comment (ydshieh) [reply src/transformers/models/opt/modeling_opt.py:1177]: Going to merge in 1h2m3s, if no further comment from Arthur :-)
